go-controller/pkg/clustermanager/userdefinednetwork/template/net-attach-def-template.go

```diff
@@ -185,10 +206,43 @@ func renderCNINetworkConfig(networkName, nadName string, spec SpecGetter) (map[s
 	if netConfSpec.AllowPersistentIPs {
 		cniNetConf["allowPersistentIPs"] = netConfSpec.AllowPersistentIPs
 	}
-
+	if netConfSpec.PhysicalNetworkName != "" {
+		cniNetConf["physicalNetworkName"] = netConfSpec.PhysicalNetworkName
+	}
+	if len(netConfSpec.ExcludeSubnets) > 0 {
+		cniNetConf["excludeSubnets"] = netConfSpec.ExcludeSubnets
+	}
+	if netConfSpec.VLANID != 0 {
+		cniNetConf["vlanID"] = netConfSpec.VLANID
+	}
 	return cniNetConf, nil
 }
 
+func validateLocalnetConfig(cfg userdefinednetworkv1.LocalnetConfig) error {
+	if cfg.Role == userdefinednetworkv1.NetworkRolePrimary {
+		return fmt.Errorf("role can be secondary only")
+	}
+	if cfg.PhysicalNetworkName == "" {
+		return fmt.Errorf("physicalNetworkName cannot be empty")
+	}
```

Comment:
why dont we validate this in CEL instead of here?

Reply:
We do have CEL validations for it at the API level https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5010/files#diff-073a0bb9fa1b183caf787b824a1bddd130ec004579ff07082596b7426d4c9b8cR111-R116 
I added similar validations on the controller side for completeness, and to prevent form creating NAD with invalid spec in case kube-apiserver validation service is down, although its not likely I think its possible.
Make sense?

Reply:
yeah i dont think it is possible